mitmproxy-rs/src/server/tun.rs

```diff
@@ -59,6 +62,24 @@ pub fn create_tun_interface(
     }
     #[cfg(not(target_os = "linux"))]
     Err(pyo3::exceptions::PyNotImplementedError::new_err(
-        "TUN proxy mode is only available on Linux",
+        unavailable_reason(),
     ))
 }
+
+/// Determines the reason for unavailability of TUN proxy mode
+///
+/// Returns
+/// - `String`: reason for unavailability
+/// - `None`: if available or reason unknown for unavailability
```

Comment:
These docstrings ends up in Python, so we should use Python syntax (str and not String). Let's also shorten this a bit.


```suggestion
/// Returns a `str` describing why tun mode is unavailable, or `None` if TUN mode is available.
/// 
/// Reasons for unavailability may be an unsupported platform, or missing privileges.
```